scripts/build.sh

```diff
@@ -344,7 +344,7 @@ if [ "$ROOT_SOL" = "kernelsu" ]; then
     elif [ "$ARCH" = "arm64" ]; then
         mv "$WORK_DIR/kernelsu/Image" "$WORK_DIR/kernelsu/kernel"
     fi
-    KSU_APP_DIR="../common/system/app/KernelSU"
+    KSU_APP_DIR="../common/system/data-app/KernelSU"
```

Comment:
Never heard about this folder

Reply:
There is no guarantee that it will work on all Android devices, but it is certainly used on some devices such as Xiaomi.

Reply:
If it did not work, I try revert.

Reply:
As you can see, it did not work.

Reply:
And use `app` didn't work too. KernelSU not installed

Reply:
> And use `app` didn't work too. KernelSU not installed

![image](https://github.com/YT-Advanced/WSA-Script/assets/52069677/c0547975-c9ad-48a6-8211-00d86d7eb82d)

Reply:
Did this problem solved ?

Reply:
No,
Cannot find /syste/app/KernelSU/

Reply:
I'm thinking about edit init.rc. You can see it here:
https://medium.com/@simo.mut105/ship-precompiled-apks-in-your-custom-rom-8d9a40dabdd6

Reply:
A possible problem with this is whether the KernelSU manager can be updated at the same time while retaining the WSA user data.
Due to the structure of the WSA system, APKs cannot be installed from the Windows side without ADB, so APKs must be placed in the system image on the Android side.

Reply:
As the BlissOS developer states, it is certain that the APK must be placed somewhere in the system, so if you can get the status of Android system update completion, you should be able to do it without problems.

~~It would only be a matter if KernelSU removing this extra functionality...~~

Reply:
> As the BlissOS developer states, it is certain that the APK must be placed somewhere in the system, so if you can get the status of Android system update completion, you should be able to do it without problems.
> 
> ~It would only be a matter if KernelSU removing this extra functionality...~

You can place it somewhere else as long as the init script can be able to execute `pm install`, it doesn't matter

Reply:
We know that.
My concern is another issue.

Reply:
> We know that. My concern is another issue.

If it's related to "how to check if the image has been updated" then you can choose props like :
- Datetime build
- Incremental version
and some more

or you can create a property on your own. Unfortunately I can't find a prop that determine if you replace a new system images yet but still keep the data intact

Reply:
@s1204IT Any idea with `test-fix` branch ?

Reply:
https://github.com/YT-Advanced/WSA-Script/compare/6b66d3a...test-fix

1. I think you need to check how Android was booted other than `wait 30`.
2. It should not be placed in `priv-app`.
   That is exactly what you should put in a directory that is not recognized as a system app, at least not as a system app, such as `data-app` mentioned earlier.
3. This is optional, but it would be easier to understand if the file name is not `notfirstrun` but rather something that indicates completion, such as `ksu_completed`.

Reply:
Perhaps something like this could be used.
https://github.com/Howard20181/magisk-modules-disabler/blob/main/util_func.sh#L32-L46